### PR TITLE
STOSB-1647 [FE] Agent- STF single w/o C&D- Build Page 17- 'What is the total market value of the securities?'

### PR DESCRIPTION
**Pull request by tharunprasath**

### CODE REVIEW

File: app/uk/gov/hmrc/securitiestransferchargefrontend/forms/stf/agents/TotalMarketValueFormProvider.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  def apply(): Form[BigDecimal] =
+    Form(
+      "value" -> currency(
+        "agent.totalMarketValue.error.required",
+        "agent.totalMarketValue.error.invalidNumeric",
+        "agent.totalMarketValue.error.nonNumeric",
+        "agent.totalMarketValue.error.negative",
+      )
+      .verifying(maximumCurrency(Int.MaxValue, "agent.totalMarketValue.error.aboveMaximum"))
```
Comment (ttiyemba) [inline app/uk/gov/hmrc/securitiestransferchargefrontend/forms/stf/agents/TotalMarketValueFormProvider.scala]: max value is 999,999,999 and this needs to be bigdecimal instead of int

File: test/controllers/stf/agents/single/TotalMarketValueControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+  val validAnswer = 0
+
+  lazy val totalMarketValueRoute: String = agentSingleRoutes.TotalMarketValueController.onPageLoad(NormalMode).url
+
+  "TotalMarketValue Controller" - {
+
+    "must return OK and the correct view for a GET" in {
+
+      val application = applicationBuilder(userAnswers = Some(emptyUserAnswers))
```
Comment (ttiyemba) [inline test/controllers/stf/agents/single/TotalMarketValueControllerSpec.scala]: could you also pass the agent affinity

File: test/controllers/stf/agents/single/TotalMarketValueControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+import uk.gov.hmrc.securitiestransferchargefrontend.views.html.stf.agents.single.TotalMarketValueView
+
+import scala.concurrent.Future
+
+class TotalMarketValueControllerSpec extends SpecBase with MockitoSugar {
+
+  val formProvider = new TotalMarketValueFormProvider()
+  val form: Form[BigDecimal] = formProvider()
+
+  lazy val totalMarketValuePageRoute: String = agentSingleRoutes.TotalMarketValueController.onPageLoad(NormalMode).url
```
Comment (ttiyemba) [inline test/controllers/stf/agents/single/TotalMarketValueControllerSpec.scala]: totalMarketValuePageRoute and totalMarketValueRoute have the same value

File: test/forms/stf/agents/TotalMarketValueFormProviderSpec.scala [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+class TotalMarketValueFormProviderSpec extends CurrencyFieldBehaviours {
+
+  private val form = new TotalMarketValueFormProvider()()
+
+  ".value" - {
+
+    val fieldName = "value"
+
+    val minimum = 0
+    val maximum = Int.MaxValue
```
Comment (ttiyemba) [inline test/forms/stf/agents/TotalMarketValueFormProviderSpec.scala]: this needs to be big decimal instead of int

File: test/forms/stf/agents/TotalMarketValueFormProviderSpec.scala [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      form,
+      fieldName,
+      maximum,
+      FormError(
+        fieldName,
+        "agent.totalMarketValue.error.aboveMaximum",
+        Seq(currencyFormat(maximum))
+      )
+    )
+
```
Comment (ttiyemba) [inline test/forms/stf/agents/TotalMarketValueFormProviderSpec.scala:67]: could you please add a test for negative values

File: app/uk/gov/hmrc/securitiestransferchargefrontend/navigation/stf/agents/ForwardRoutes.scala [hunk 6]
```diff
@@ -71,6 +71,14 @@ class ForwardRoutes(answerPersistenceService: AnswerPersistenceService,
     }
     case TaxRatePage => userAnswers => dataRequired(TaxRatePage, userAnswers, agentSingleRoutes.WhatTypeOfSecuritiesController.onPageLoad(NormalMode))
     case OtherSecuritiesTypePage => userAnswers => dataRequired(OtherSecuritiesTypePage, userAnswers, agentSingleRoutes.AmountPaidForSecuritiesController.onPageLoad(NormalMode))
+    case AmountPaidForSecuritiesPage => userAnswers =>
+      dataDependent(ConnectedPersonsPage, userAnswers) { isConnected =>
+        if (isConnected)
+          agentSingleRoutes.TotalMarketValueController.onPageLoad(NormalMode)
```
Comment (ttiyemba) [inline app/uk/gov/hmrc/securitiestransferchargefrontend/navigation/stf/agents/ForwardRoutes.scala:77]: do we have a test for this in the nav spec?

**Comment by ttiyemba** [comment]
could you also add the missing summary file

**Review by ttiyemba — APPROVED** [review]
